### PR TITLE
Cast function

### PR DESCRIPTION
**Pull request by unnonouno**
I think this function is required to use fp16.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/array_tests/test_cast.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        self.check_forward(self.x)
+
+    @attr.gpu
+    def test_forward_gpu(self):
+        self.check_forward(cuda.to_gpu(self.x))
+
+    def check_backward(self, x_data, g_data):
+        func = functions.Cast(self.out_type)
+        gradient_check.check_backward(
+            func, x_data, g_data, eps=2.0 ** 5, atol=1e-3, rtol=1e-3)
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/array_tests/test_cast.py]: `eps=2.0 ** 5` is collect?

**Comment by okuta** [comment]
LGTM except comment.

**Comment by unnonouno** [comment]
I fixed the test.

**Comment by okuta** [comment]
Please add `testing.run_module(__name__, __file__)` to end of test file.

**Comment by unnonouno** [comment]
i fixe it

**Comment by okuta** [comment]
OK!

**Pull request merged**